cnd/src/asset/bitcoin.rs

```diff
@@ -0,0 +1,55 @@
+use bitcoin::{
+    util::amount::{Denomination, ParseAmountError},
+    Amount,
+};
+use std::fmt;
+
+#[derive(Debug, Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub struct Bitcoin(Amount);
+
+impl Bitcoin {
+    pub fn from_sat(sat: u64) -> Bitcoin {
+        Bitcoin(Amount::from_sat(sat))
+    }
+
+    pub fn as_sat(self) -> u64 {
+        Amount::as_sat(self.0)
+    }
+
+    pub fn from_btc(btc: f64) -> Result<Bitcoin, ParseAmountError> {
+        Amount::from_btc(btc).map(Bitcoin)
+    }
+
+    pub fn from_str_in(
```

Comment:
I see only two usages of this, both of them use `Satoshi` as the `Denomonation`.

I think we could get rid of this if we require the caller to parse their string as `u64` and then just use the `from_sat` method.